chef/cookbooks/dhcp/providers/subnet.rb

```diff
@@ -13,11 +13,20 @@
 # limitations under the License.
 #
 
+require "ipaddr"
+
 action :add do
+  if IPAddr.new(new_resource.network["subnet"]).ipv6?
```

Comment:
I've seen this sort of thing in a number of places. I wonder if it might be cleaner to use an ip_version variable to store 4 vs. 6, and then you can simply use ip_version as a suffix and do away with some if/else blocks and make any file names explicit (ie subnet4.conf.erb vs. subnet.conf.erb)

Reply:
The way that crowbar updates and removes subnets from the list does make we worry about it from an upgrade perspective. I.e an existing env with subnet_list.conf with a bunch of subnets and we suddently change it. Will only "new" subnet source config lines be added.

I'll re-roll 1/2 way between. Leave ipv4 and subnet_list.conf and use subnet6_list.conf. But change to just adding a 6 or not. It does look much cleaner. Though ip_version = '' on ipv4 but. still cleaner then it was.